spec/addressable/uri_spec.rb

```diff
@@ -5515,6 +5515,47 @@ def to_s
   end
 end
 
+describe Addressable::URI, "When set the basename " do
+  it "should have the correct basename: 'index.html'" do
+    uri = Addressable::URI.parse("http://test.com")
+    uri.basename = "index.html"
+
+    expect(uri.basename).to eq("index.html")
+    expect(uri.to_s).to eq("http://test.com/index.html")
+  end
+end
+
+describe Addressable::URI, "When change the basename " do
+  it "should have the new basename: 'item'" do
+    uri = Addressable::URI.parse("http://test.com/some/path/index.html")
+
+    expect(uri.basename = "item").to eq("item")
+    expect(uri.to_s).to eq("http://test.com/some/path/item")
+  end
+end
+
+describe Addressable::URI, "After assignment the basename " do
+  it "should have the correct basename: 'subcat_1' "\
+    "when the current basename at the end contains a slash" do
+
+    url = "http://test.com/sections/categories/sub_categories/"
```

Comment:
I believe there are already `describe` blocks w/ paths containing trailing slashes.

Reply:
Also it's not a URL when it's a `String` and a URI once it's been parsed. Please avoid this variable naming scheme. If you have to do this, use `uri_string` and `uri`.

Reply:
yes, you are right, I'm sorry this is my inattention, here was a myself checked, in particular this part of regexp: `(\/)?\z`
it's redundant, i'l remove it, thank you